lib/commands/migrate.js

```diff
@@ -1,42 +1,81 @@
 "use strict";
 
 const fs = require("fs");
+const path = require("path");
 const chalk = require("chalk");
 const diff = require("diff");
 const inquirer = require("inquirer");
 const PLazy = require("p-lazy");
 const Listr = require("listr");
 
-const validate = require("webpack").validate;
-const WebpackOptionsValidationError = require("webpack")
-	.WebpackOptionsValidationError;
+const { validate } = require("webpack");
+const { WebpackOptionsValidationError } = require("webpack");
 
 const runPrettier = require("../utils/run-prettier");
 
 /**
-*
-* Runs migration on a given configuration using AST's and promises
-* to sequentially transform a configuration file.
-*
-* @param {String} currentConfigPath - Location of the configuration to be migrated
-* @param {String} outputConfigPath - Location to where the configuration should be written
-* @param {Object} options - Any additional options regarding code style of the written configuration
+ *
+ * Runs migration on a given configuration using AST's and promises
+ * to sequentially transform a configuration file.
+ *
+ * @param {Array} args - Migrate arguments such as input and
+ * output path
+ * @returns {Function} Runs the migration using the 'runMigrate'
+ * function.
+ */
 
-* @returns {Promise} Runs the migration using a promise that will throw any errors during each transform
-* or output if the user decides to abort the migration
-*/
+module.exports = function migrate(...args) {
+	const filePaths = args ? args.slice(3) : null;
+	if (!filePaths.length) {
+		throw new Error("Please specify a path to your webpack config");
+	}
+	const currentConfigPath = path.resolve(process.cwd(), filePaths[0]);
+	let outputConfigPath;
+	if (!filePaths[1]) {
+		return inquirer
+			.prompt([
+				{
+					type: "confirm",
+					name: "confirmPath",
+					message:
+						"Migration output path not specified. " +
+						"Do you want to use your existing webpack " +
+						"configuration?",
+					default: "Y"
+				}
+			])
+			.then(ans => {
+				if (!ans["confirmPath"]) {
+					console.log(chalk.red("✖ ︎Migration aborted due no output path"));
+					process.exitCode = 1;
```

Comment:
I think without `return`, this might continue with migration.

Reply:
no

Reply:
![screenshot_20180401_165304](https://user-images.githubusercontent.com/5961873/38175244-0881e628-35f7-11e8-9512-3ee9124c2101.png)

Reply:
`Migration aborted due no output path`

Reply:
https://www.nczonline.net/blog/2014/02/04/maintainable-node-js-javascript-avoid-process-exit/

Reply:
That's true but `exitCode` won't exit the process as it doesn't call the `exit()` method implicitly. To exit the process we'll need to do:
```js
process.exitCode = 1;
process.exit(); // this will exit with exit code 1
```
or
```js
process.exit(1); // this will also exit with exit code 1
```
or add a `return;`.

Without exiting, the process will continue and do the migration even after the chalk error log.

Reply:
@dhruvdutt is right, we need to abort from the chain of promises. But `return` won't help. I'd suggest to throw an error and handle it inside the `catch`